File: sdkcreationmw/sdkruntimes/MIDP/DebugAgent/debugagent_orig/src/seislave/commands/HandshakeCommand.cpp
/*
* Copyright (c) 2004 Nokia Corporation and/or its subsidiary(-ies).
* All rights reserved.
* This component and the accompanying materials are made available
* under the terms of "Eclipse Public License v1.0"
* which accompanies this distribution, and is available
* at the URL "http://www.eclipse.org/legal/epl-v10.html".
*
* Initial Contributors:
* Nokia Corporation - initial contribution.
*
* Contributors:
*
* Description: 
*
*/


#include "SeiCommands.h"
#include "TcpServerHandler.h"


CHandshakeCommand* CHandshakeCommand::NewL(
    CDriver* aDriver, 
    CSocketReader* aReader,
    CSocketWriter* aWriter,
    CTcpServerHandler* aHandler)
    {
    CHandshakeCommand* object = new (ELeave) CHandshakeCommand(aDriver, aReader, aWriter, aHandler);
    CleanupStack::PushL(object);
    object->ConstructL();
    CleanupStack::Pop(object);
    return object;
    }

CHandshakeCommand::CHandshakeCommand(
    CDriver* aDriver, 
    CSocketReader* aReader,
    CSocketWriter* aWriter,
    CTcpServerHandler* aHandler)
    : 	CSeiCommand(aDriver, aReader, aWriter), 
    	iPinCode(NULL), iHandler(aHandler)
    {
    }

CHandshakeCommand::~CHandshakeCommand()
    {
    if ( iPinCode != NULL )
    	{
    	delete iPinCode;
    	iPinCode = NULL;
    	}
    }

void CHandshakeCommand::ConstructL()
    {
    }

void CHandshakeCommand::DoReadL()
    {
	iDriver->Log(_L("Authenticating"));
	iReader->ReadUtfL(iPinCode);
    }

void CHandshakeCommand::DoExecuteL()
    {
    if ( iDriver->IsUsePin() )
    	{
    	HBufC* pin = iDriver->GetPinCode();
    	iResult = pin->Compare(*iPinCode);
    	}
    else 
    	{
    	iResult = KErrNone;	
    	}
    iHandler->SetAuthenticated(iResult==KErrNone);
    }

void CHandshakeCommand::DoWriteL()
    {
    if ( iResult != KErrNone )
    	{
		iDriver->Log(_L("Authentication failed."));
    	}
    else
    	{
		iDriver->Log(_L("Authentication ok."));
    	}
    iWriter->WriteIntL(iResult);
    }

File: sdkcreationmw/sdkruntimes/MIDP/nei/src/com/symbian/tools/j2me/sei/emulator/session/commands/HandshakeCommand.java
/*
* Copyright (c) 2003 - 2004 Nokia Corporation and/or its subsidiary(-ies).
* All rights reserved.
* This component and the accompanying materials are made available
* under the terms of "Eclipse Public License v1.0"
* which accompanies this distribution, and is available
* at the URL "http://www.eclipse.org/legal/epl-v10.html".
*
* Initial Contributors:
* Nokia Corporation - initial contribution.
*
* Contributors:
*
* Description: 
*
*/

package com.symbian.tools.j2me.sei.emulator.session.commands;

import java.io.*;
import com.symbian.vm.*;

public class HandshakeCommand extends SessionCommand{


  //
  // Constants
  //


  /**
   * Handshake request identifier
   */
  private static final int HANDSHAKE_CMD_ID = 8;


  //
  // Members
  //

  /**
   * Challenge is set by the user and sent at the beginning of the session
   */
  private String iChallenge;


  //
  // Life cycle
  //

  /**
   * Default constructor.
   *
   * @param aChallenge challenge string used to authenticate the session
   */
  public HandshakeCommand(String aChallenge){
    super(HANDSHAKE_CMD_ID);
    iChallenge = aChallenge;
  }

  /**
   * Dump the object to the output stream
   *
   * @param aDos Data output stream
   *
   * @throws IOException if some error occured
   */
  public void dump(DataOutputStream aDos) throws IOException{
    super.dump(aDos);
    aDos.writeUTF(iChallenge);
    aDos.flush();
  }

}
